pkg/providers/k8s/endpoint/base.go

```diff
@@ -131,3 +131,23 @@ func (c *baseEndpointController) syncEndpoint(ctx context.Context, ep kube.Endpo
 	}
 	return nil
 }
+
+func (c *baseEndpointController) syncEmptyEndpoint(ctx context.Context, ep kube.Endpoint) error {
+	namespace, err := ep.Namespace()
+	if err != nil {
+		return err
+	}
+	svcName := ep.ServiceName()
+	log.Infow("The syncEndpoint, service has been deleted, try to update upstream",
```

Comment:
```suggestion
	log.Debugw("The service has been deleted, try to delete upstream relation",
```
Potential frequently called function trace logging should be debug level.